src/openrct2/windows/object_load_error.c

```diff
@@ -0,0 +1,326 @@
+
+static rct_window_event_list window_object_load_error_events = {
+    window_object_load_error_close,
+    window_object_load_error_mouseup,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    window_object_load_error_update,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    window_object_load_error_scrollgetsize,
+    window_object_load_error_scrollmousedown,
+    NULL,
+    window_object_load_error_scrollmouseover,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    NULL,
+    window_object_load_error_paint,
+    window_object_load_error_scrollpaint
+};
+
+rct_object_entry * * invalid_entries = NULL;
+sint32 highlighted_index = -1;
+utf8* file_path = NULL;
+
+/**
+*  Returns an rct_string_id that represents an rct_object_entry's type.
+*
+*  Could possibly be moved out of the window file if other
+*  uses exist and a suitable location is found.
+*/
+static rct_string_id get_object_type_string(const rct_object_entry *entry)
+{
+    rct_string_id result;
+    uint8 objectType = entry->flags & 0x0F;
+    switch (objectType) {
+        case OBJECT_TYPE_RIDE:
+            result = STR_OBJECT_SELECTION_RIDE_VEHICLES_ATTRACTIONS;
+            break;
+        case OBJECT_TYPE_SMALL_SCENERY:
+            result = STR_OBJECT_SELECTION_SMALL_SCENERY;
+            break;
+        case OBJECT_TYPE_LARGE_SCENERY:
+            result = STR_OBJECT_SELECTION_LARGE_SCENERY;
+            break;
+        case OBJECT_TYPE_WALLS:
+            result = STR_OBJECT_SELECTION_WALLS_FENCES;
+            break;
+        case OBJECT_TYPE_BANNERS:
+            result = STR_OBJECT_SELECTION_PATH_SIGNS;
+            break;
+        case OBJECT_TYPE_PATHS:
+            result = STR_OBJECT_SELECTION_FOOTPATHS;
+            break;
+        case OBJECT_TYPE_PATH_BITS:
+            result = STR_OBJECT_SELECTION_PATH_EXTRAS;
+            break;
+        case OBJECT_TYPE_SCENERY_SETS:
+            result = STR_OBJECT_SELECTION_SCENERY_GROUPS;
+            break;
+        case OBJECT_TYPE_PARK_ENTRANCE:
+            result = STR_OBJECT_SELECTION_PARK_ENTRANCE;
+            break;
+        case OBJECT_TYPE_WATER:
+            result = STR_OBJECT_SELECTION_WATER;
+            break;
+        case OBJECT_TYPE_SCENARIO_TEXT:
+            result = STR_OBJECT_SELECTION_SCENARIO_DESCRIPTION;
+            break;
+        default:
+            result = STR_UNKNOWN_OBJECT_TYPE;
+    }
+    return result;
+}
+
+/**
+*  Returns a newline-separated string listing all object names.
+*  Used for placing all names on the clipboard.
+*/
+static utf8* combine_object_names(rct_window *w)
+{
+    if (w->no_list_items > OBJECT_ENTRY_COUNT || w->no_list_items == 0) {
+        // Something's gone wrong, this shouldn't happen
+        // We don't want to allocate stupidly large amounts of memory
+        // for no reason, so bail
+        return NULL;
+    }
+    utf8* buffer;
+    utf8* line_sep = platform_get_line_separator();
```

Comment:
There is already a line separator `#define` somewhere that you can use.
Also this needs to be `freed` as your function allocates memory.